conans/client/generators/make.py

```diff
@@ -0,0 +1,118 @@
+                     ("DEFINES", self.assignment_append, cpp_info.defines),
+                     ("CFLAGS", self.assignment_append, cpp_info.cflags),
+                     ("CPPFLAGS", self.assignment_append, cpp_info.cppflags),
+                     ("SHAREDLINKFLAGS", self.assignment_append, cpp_info.sharedlinkflags),
+                     ("EXELINKFLAGS", self.assignment_append, cpp_info.exelinkflags)]
+
+        return [self.create_makefile_var_pkg(var_name, pkg_name, operator, info)
+                for var_name, operator, info in vars_info]
+
+    def create_combined_content(self):
+        content = []
+        for var_name in self.all_dep_vars():
+            content.append(self.create_makefile_var_global(var_name, self.assignment_append,
+                                                           self.create_combined_var_list(var_name)))
+        return content
+
+    def create_combined_var_list(self, var_name):
+        make_vars = []
+        for pkg_name, _ in self.deps_build_info.dependencies:
+            pkg_var = self.create_makefile_var_name_pkg(var_name, pkg_name)
+            make_vars.append("$({pkg_var})".format(pkg_var=pkg_var))
+        return make_vars
+
+    def create_makefile_var_global(self, var_name, operator, values):
+        make_var = [self.create_makefile_var_name_global(var_name)]
+        make_var.extend(self.create_makefile_var_common(operator, values))
+        return make_var
+
+    def create_makefile_var_pkg(self, var_name, pkg_name, operator, values):
+        make_var = [self.create_makefile_var_name_pkg(var_name, pkg_name)]
+        make_var.extend(self.create_makefile_var_common(operator, values))
+        return make_var
+
+    def create_makefile_var_common(self, operator, values):
+        return [operator, self.makefile_line_continuation, self.create_makefile_var_value(values),
+                self.makefile_newline]
+
+    @staticmethod
+    def create_makefile_var_name_global(var_name):
+        return "CONAN_{var}".format(var=var_name).upper()
+
+    @staticmethod
+    def create_makefile_var_name_pkg(var_name, pkg_name):
+        return "CONAN_{var}_{lib}".format(var=var_name, lib=pkg_name).upper()
+
+    def create_makefile_var_value(self, values):
+        formatted_values = [value.replace("\\", "/") for value in values]
+        return self.makefile_line_continuation.join(formatted_values)
+
+    @staticmethod
+    def all_dep_vars():
+        return ["rootpath", "sysroot", "include_paths", "lib_paths", "bin_paths", "build_paths",
+                "res_paths", "libs", "defines", "cflags", "cppflags", "sharedlinkflags",
+                "exelinkflags"]
+
+    @property
+    def makefile_newline(self):
+        return "\n"
+
+    @property
+    def makefile_line_continuation(self):
+        return " \\\n"
+
+    @property
+    def assignment_if_absent(self):
+        return " ?= "
+
+    @property
+    def assignment_append(self):
+        return " += "
```

Comment:
Why properties and not plain attributes?

Reply:
I left it as it was. I think it is done that way to prevent any change in the content